src/Features/Core/Portable/CodeRefactoringHelpers.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class CodeRefactoringHelpers
+    {
+        public static Task<bool> RefactoringSelectionIsValidAsync(
+            Document document, TextSpan selection, SyntaxNode node, CancellationToken cancellation)
+        {
+            return RefactoringSelectionIsValidAsync(document, selection, node, ImmutableArray<SyntaxNode>.Empty, cancellation);
+        }
+
+        public static async Task<bool> RefactoringSelectionIsValidAsync(
+            Document document, TextSpan selection, SyntaxNode node,
+            ImmutableArray<SyntaxNode> holes, CancellationToken cancellationToken)
+        {
+            if (selection.Length == 0)
+            {
+                return await RefactoringPositionIsValidAsync(
+                    document, selection.Start, node, holes, cancellationToken).ConfigureAwait(false);
+            }
+
+            // If we have selection, it needs to be selecting at least the full node. We are lenient
+            // with the selection starting somewhere between the full node-line span, and the start
+            // of the node as well as ending after the end of the node and the end of the line the
+            // node ends on.
```

Comment:
this comment may not be valid anymore.  didn't you want to support the selection going up to the end of the *previous* line to support people easily selecting by hitting 'up'?